src/lib.rs

```diff
@@ -512,11 +643,12 @@ impl GooseAttack {
             test_stop_task: None,
             task_sets: Vec::new(),
             weighted_users: Vec::new(),
-            host: None,
-            configuration,
+            defaults: GooseDefaults::default(),
+            configuration: GooseConfiguration::parse_args_default_or_exit(),
```

Comment:
Parsing the config already happens in this funcion, some lines above.

Reply:
What's the difference between the GooseAttack configuration from the arguments and the default? It is unclear to me which one should be used where

Reply:
It's orders of precedence:

- if a cli flag/option is set, we always use this
- otherwise, if a custom default is defined, we use this
- otherwise, we use Goose's base defaults

The rules are slightly more complex for something like the host name:
- if the URL itself defines a host, we use that
- otherwise, if the --host flag is set, we use that
- otherwise, if a host is defined for the task set, we use that
- otherwise, if a default host is defined, we use that
- otherwise we fail on error, as a host is required

Reply:
Also note that Gaggle's complicate things a little more.

Most flags/options an only be set on the Manager, so `set_default()` can define a default and the underlying logic only sets these defaults on the Manager, and not on the Worker (as the Worker must instead inherit these settings from the Manager).

That said, there are a few options that can only be set on the Worker, and again `set_default()` will only set these defaults on the Worker, not on the Manager.